qiskit/algorithms/linear_solvers/matrices/tridiagonal.py

```diff
@@ -0,0 +1,385 @@
+
+        Returns:
+            The number of state qubits.
+        """
+        return self._num_state_qubits
+
+    @num_state_qubits.setter
+    def num_state_qubits(self, num_state_qubits: int) -> None:
+        """Set the number of state qubits.
+
+        Note that this may change the underlying quantum register, if the number of state qubits
+        changes.
+
+        Args:
+            num_state_qubits: The new number of qubits.
+        """
+        if num_state_qubits != self._num_state_qubits:
+            self._invalidate()
+            self._num_state_qubits = num_state_qubits
+            self._reset_registers(num_state_qubits)
+
+    @property
+    def main_entry(self) -> float:
+        """Return the entry in the main diagonal."""
+        return self._main_entry
+
+    @main_entry.setter
+    def main_entry(self, main_entry: float) -> None:
+        """Set the entry in the main diagonal.
+        Args:
+            main_entry: The new entry in the main diagonal.
+        """
+        self._main_entry = main_entry
+
+    @property
+    def off_diag(self) -> float:
+        """Return the entry in the off diagonals."""
+        return self._off_diag
+
+    @off_diag.setter
+    def off_diag(self, off_diag: float) -> None:
+        """Set the entry in the off diagonals.
+        Args:
+            off_diag: The new entry in the main diagonal.
+        """
+        self._off_diag = off_diag
+
+    @property
+    def tolerance(self) -> float:
+        """Return the error tolerance"""
+        return self._tolerance
+
+    @tolerance.setter
+    def tolerance(self, tolerance: float) -> None:
+        """Set the error tolerance.
+        Args:
+            tolerance: The new error tolerance.
+        """
+        self._tolerance = tolerance
+
+    @property
+    def evo_time(self) -> float:
+        """Return the time of the evolution."""
+        return self._evo_time
+
+    @evo_time.setter
+    def evo_time(self, evo_time: float) -> None:
+        """Set the time of the evolution and update the number of Trotter steps because the error
+         tolerance is a function of the evolution time and the number of trotter steps.
+
+        Args:
+            evo_time: The new time of the evolution.
+        """
+        self._evo_time = evo_time
+        # Update the number of trotter steps. Max 7 for now, upper bounds too loose.
+        self.trotter = int(np.ceil(np.sqrt(((evo_time * np.abs(self.off_diag)) ** 3) / 2
+                                           / self.tolerance)))
+
+    @property
+    def trotter(self) -> int:
+        """Return the number of trotter steps."""
+        return self._trotter
+
+    @trotter.setter
+    def trotter(self, trotter: int) -> None:
+        """Set the number of trotter steps.
+        Args:
+            trotter: The new number of trotter steps.
+        """
+        self._trotter = trotter
+
+    @property
+    def matrix(self) -> np.ndarray:
+        """Return the matrix."""
```

Comment:
Can this be explained since the docstring does not tell you anything you cannot imply from the name of the getter. How about `returns the matrix built according to the main and off diag entries` or some such text that says what the matrix is

Reply:
Updated this and added an explicit example of such a matrix in the documentation at the beginning of the class.